internal/fs/wrappers/error_mapping.go

```diff
@@ -59,16 +60,29 @@ func errno(err error) error {
 	}
 
 	// Cannot authenticate or Permission denied.
-	// The control client API returns an RPC error code instead of googleapi code.
-	if strings.Contains(err.Error(), "oauth2: cannot fetch token") || strings.Contains(err.Error(), "PermissionDenied") {
+	if strings.Contains(err.Error(), "oauth2: cannot fetch token") {
 		return syscall.EACCES
 	}
 
-	// Translate API errors into an em errno
-	var apiErr *googleapi.Error
+	// The control client API returns an RPC error code instead of googleapi code.
+	// Currently, we only have gRPC control client APIs, so we are checking the gRPC status code.
```

Comment:
control client apis are only for hierarchial buckets? for regular buckets we dont have control client apis or are those http apis?

Reply:
control client apis are only for hierarchial buckets? --> Yes 
For regular buckets we dont have control client apis or are those http apis? --> Http/Grpc storage client APIs.